_posts/2024-14-18-asp-react-docker-compose.md

```diff
@@ -0,0 +1,270 @@
+
+In it create 3 folders and 1 file in root folder :
+#### The Project structure is:
+```plaintext
+app
+├── backend
+├── docker-compose.yml
+├── frontend
+└── nginx
+
+4 directories, 1 file
+```
+Here is the reference of project public repo in github [Source Code](https://github.com/usman-jamil/dev-docs/tree/main) of given example project asp-react-nginx-compose.
+
+### ASP.NET 8.0 and React.ts with Docker Compose
+
+This guide explains how to set up a project structure using ASP.NET, React, and Nginx within Docker containers. Below are detailed steps for configuring your development environment, including Docker installation.
+
+* For enabling Docker support in project it is important Docker is install and running in Laptop and computer.
+## Installing Docker
+
+Before you start setting up your project, you need to install Docker on your machine if it isn't already installed. Docker allows you to create, deploy, and run applications by using containers.
+
+### For Windows
+
+1. Download Docker Desktop from the [official Docker website](https://www.docker.com/products/docker-desktop).
+2. Run the installer and follow the on-screen instructions.
+3. Ensure that the "Windows Subsystem for Linux" (WSL2) feature is enabled on your machine.
+4. After installation, launch Docker Desktop and sign in or create a Docker account if prompted.
+
+### For macOS
+
+1. Download Docker Desktop for Mac from the [official Docker website](https://www.docker.com/products/docker-desktop).
+2. Open the downloaded `.dmg` file and drag the Docker icon to your Applications folder.
+3. Open Docker from your Applications to complete the installation.
+4. Docker may request your password to install a helper tool; provide your password to continue.
+
+### For Linux
+
+The installation commands can vary depending on your Linux distribution. Here's how you can install Docker on Ubuntu:
+
+1. Update your existing list of packages:
+```bash
+  $ sudo apt update
+
+  $ sudo apt install apt-transport-https ca-certificates curl software-properties-common
+
+  $ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+
+  $ sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+
+  $ sudo apt update
+
+  $ apt-cache policy docker-ce
+
+  $ sudo apt install docker-ce
+
+docker run hello-world
+```
+## Backend Configuration
+
+### Creating the ASP.NET Project
+
+Navigate into the `backend` folder and create a new ASP.NET project. You can add Docker support during the project creation or after the project has been created.
+#### During Project Creation
+1. Mark check true to Enable Docker
+2. Select Docker OS: Linux
+
+<img src="../assets/img/Screenshot1.png" width="50%" height="50%">
+
+#### After Project Creation
+
+Right-click on the project name in your IDE, then:
+
+1. Click "Add"
+2. Select "Docker Support"
+
+<img src="../assets/img/Screenshot2.png" width="50%" height="50%">
```

Comment:
Remove these lines in favor of "please see the Dockerfile inside backend folder. It contains the following contents. Notice how the API is exposed on port 8080 and port 8081"

Reply:
done